### ISSUE TITLE
[Assignment] Results, Implications, Challenges

### ISSUE THREAD

**Issue by Yuki-bach** [Yuki-bach/marketing#62]
## Due date
Apr 14 

## ToDo
- Feature Engineering
  - [x] Create RFM cluster using 3 features.
  - [x] Sampling 5000 records to compare 3 models. 
  - [ ] Explain how we created each feature in the Methodology section of our final paper.
- Clustering
  - k-means, DBSCAN
    - [x] Get Davies-Bouldin index.
  - Agglomerative
    - [ ] Standardize and normalize training data and compare the scores.
    - [ ] Interpret the dendrograms.
    - [ ] Find the reason why cosine and correlation metrics failed to cluster.
- XGBoosting
  - [x] Use the 3-4 months for training data.
- Sales Analysis
  - [x] Create 3 models to predict the future sales per cluster. (Simple Regression, Polynomial Regression, ARIMA) 
- Q1: Repeater Analysis
  - [x] Change the way of SMOTE application
  - [x] Consider additional performance metrics like ROC-curve, Precision, Recall, F1-Score, and AUC.
  - [x] Explore feature importance within predictive models to identify key drivers behind repeat customer behavior.
  - [ ] Data Preprocessing
- Customer Reviews (NLP)
  - [x] Create word cloud to visualize review text per cluster.

**Comment by Yuki-bach** [comment]
We submitted the result assignment.
